server/src/actors/events.rs

```diff
@@ -125,6 +142,162 @@ impl EventProcessor {
+            .unwrap_or_default();
+
+        let mut breadcrumbs2 = breadcrumbs2
+            .as_ref()
+            .and_then(Self::message_pack_to_annotated_breadcrumbs)
+            .unwrap_or_default();
+
+        let timestamp1 = breadcrumbs1
+            .iter()
+            .rev()
+            .find_map(|breadcrumb| breadcrumb.value().and_then(|b| b.timestamp.value()));
+
+        let timestamp2 = breadcrumbs2
+            .iter()
+            .rev()
+            .find_map(|breadcrumb| breadcrumb.value().and_then(|b| b.timestamp.value()));
+
+        if timestamp1 > timestamp2 {
+            std::mem::swap(&mut breadcrumbs1, &mut breadcrumbs2);
+        }
+
+        let max_length = std::cmp::max(breadcrumbs1.len(), breadcrumbs2.len());
+
+        breadcrumbs1.extend(breadcrumbs2);
+
+        if breadcrumbs1.len() > max_length {
+            // keep only the last max_length elements from the vectors
+            breadcrumbs1.drain(0..(breadcrumbs1.len() - max_length));
+        }
+
+        if let Some(evt) = evt.value_mut() {
+            evt.breadcrumbs = Annotated::new(Values {
+                values: Annotated::new(breadcrumbs1.into()),
+                other: Object::default(),
+            });
+        }
+    }
+
+    /// Writes a placeholder to indicate that this event has an associated minidump.
+    fn write_minidump_placeholder(evt: &mut Annotated<Event>) {
+        let evt = evt.get_or_insert_with(Event::default);
+
+        let platform = evt.platform.value_mut();
+        *platform = Some("native".to_string());
+
+        let level = evt.level.value_mut();
+        *level = Some(Level::Fatal);
+
+        let exceptions = evt
+            .exceptions
+            .value_mut()
+            .get_or_insert_with(Values::<Exception>::default)
+            .values
+            .value_mut()
+            .get_or_insert_with(|| vec![]);
+
+        exceptions.clear(); // clear previous errors if any
+
+        exceptions.push(Annotated::new(Exception {
+            ty: Annotated::new("Minidump".to_string()),
+            value: Annotated::new(JsonLenientString("Invalid Minidump".to_string())),
+            mechanism: Annotated::new(Mechanism {
+                ty: Annotated::from("minidump".to_string()),
+                handled: Annotated::from(false),
+                synthetic: Annotated::from(true),
+                ..Mechanism::default()
+            }),
+            ..Exception::default()
+        }));
+    }
+
+    /// merge form data (from a multipart request into the event
+    fn event_from_minidump_data(
+        form_data: Item,
+        event_data_mps: Option<Item>,
+        breadcrumbs1: Option<Item>,
+        breadcrumbs2: Option<Item>,
+    ) -> Annotated<Event> {
+        // if we have a "sentry" entry it should be a straight json.. merge this into
+        if form_data.ty() != ItemType::FormData
+            || form_data.content_type() != Some(&ContentType::Text)
```

Comment:
Shouldn't this be Json?

```suggestion
            || form_data.content_type() != Some(&ContentType::Json)
```

Reply:
Form data is no longer JSON.
Now I use a streaming serializer to avoid creating a temporary vector of string tuples.
The streaming serializer serializes a sequence of  key-value vectors. 
So the contents of the  FormData item looks like this:
```json
["key1","Value1"]["key2","value2"]["key3","value3"]
```
Which is **NOT** a json object but as sequence of json arrays.

Reply:
Okay, let's leave JSON but please add a comment about this. We can come up with a better content type later.